### PR TITLE
Ensure custom enum symbol names in intellisense suggestions are escaped

### PR DESCRIPTION
**Pull request by lesaltzm**
As none of our builtin enum names are escaped, we have a gap around suggestions, where if a custom enum is created, it isn't properly escaped in the intellisense suggestions. This only applies to canvas, for PCF custom enums. Every other use case is on Option Sets.

### CODE REVIEW

**Review by Alex-MSFT — APPROVED** [review]

**Review by Ananya-Gupta24 — DISMISSED** [review]

**Comment by LucGenetier** [comment]
✅ No public API change.

**Comment by LucGenetier** [comment]
✅ No public API change.

**Review by MikeStall — DISMISSED** [review]

**Comment by Shpakh-MSFT** [comment]
>                 var enumName = enumInfo.EntityName.Value;

var enumName = TexlLexer.EscapeName(enumInfo.EntityName.Value) instead?

---
Refers to: src/libraries/Microsoft.PowerFx.Core/Texl/Intellisense/IntellisenseHelper.cs:758 in 5780456. [](commit_id = 5780456b4d2d7e12a96a6ccc1b5cc0e40fd7a9d6, deletion_comment = False)

**Comment by Shpakh-MSFT** [comment]
>                 AddSuggestion(intellisenseData, unqualifiedEnum, SuggestionKind.Enum, SuggestionIconKind.Other, enumType, requiresSuggestionEscaping: false);

We are not going to have unqualified enums any more and we will need to clean this code up. So this doesn't matter but for consistency, should we be escaping unqualifiedenum as well?

---
Refers to: src/libraries/Microsoft.PowerFx.Core/Texl/Intellisense/IntellisenseHelper.cs:798 in 5780456. [](commit_id = 5780456b4d2d7e12a96a6ccc1b5cc0e40fd7a9d6, deletion_comment = False)

File: src/tests/Microsoft.PowerFx.Core.Tests/IntellisenseTests/SuggestTest.cs [hunk 1]
```diff
@@ -235,6 +237,26 @@ public void TestSuggestEmptyAll(string expression, params string[] expectedSugge
             Assert.Equal(expectedSuggestions, actualSuggestions);
         }
 
+        [Fact]
+        public void TestSuggestEscapedEnumName()
+        {
+            var enumStoreBuilder = new EnumStoreBuilder();
+            enumStoreBuilder.TestOnly_WithCustomEnum(new EnumSymbol(
+                new DName("Name That.Requires!escaping"),
```
Comment (Shpakh-MSFT) [inline src/tests/Microsoft.PowerFx.Core.Tests/IntellisenseTests/SuggestTest.cs:245]: >! [](http://example.com/codeflow?start=45&length=1)

'.' please

"Name That.Requires.escaping"

Comment (lesaltzm) [reply src/tests/Microsoft.PowerFx.Core.Tests/IntellisenseTests/SuggestTest.cs:245]: No, the ! is also a character that requires escaping. In this case it's not used as the ! operator, just as a character in an identifier.

Comment (Shpakh-MSFT) [reply src/tests/Microsoft.PowerFx.Core.Tests/IntellisenseTests/SuggestTest.cs:245]: Oh sorry my bad. I didn't see that.

**Review by Shpakh-MSFT — DISMISSED** [review]
:shipit:

**Comment by LucGenetier** [comment]
✅ No public API change.

**Comment by LucGenetier** [comment]
✅ No public API change.

**Review by Ananya-Gupta24 — APPROVED** [review]

**Review by Shpakh-MSFT — APPROVED** [review]
:shipit:

**Comment by LucGenetier** [comment]
✅ No public API change.

**Pull request merged**